### ISSUE TITLE
Named unapply selectors accept random names

### ISSUE THREAD

**Issue by KacperFKorban** [scala/scala3#22192]
## Compiler version

91063dd9939ba26f30a538889162b6832cfc2c19

## Minimized code

```Scala
//> using scala 3.nightly

import scala.language.experimental.namedTuples

case class City(name: String, population: Int)

def getCityInfo(city: City) =
  city match
    case City(iam = n, confused = p) => s"[City] $n has a population of $p !!!!!!!!!!"

@main
def main =
  val city = City(name = "New York", population = 8_000_000)
  println(getCityInfo(city))
```

## Output

```scala
[City] New York has a population of 8000000 !!!!!!!!!!
```

## Expectation

Compilation error, that `iam` and `confused` aren't fields of the class `City`

**Comment by KacperFKorban** [comment]
The problem is most likely that the `tryEither` [here](https://github.com/scala/scala3/blob/91063dd9939ba26f30a538889162b6832cfc2c19/compiler/src/dotty/tools/dotc/typer/Applications.scala#L180) swallows the error.

**Comment by KacperFKorban** [comment]
@rochala I don't think that it's the same issue. This one is related to the compiler not checking the names of the fields, not the combinations of named/unnamed patterns.
Your second explanation point says something that seems similar to this issue, but you don't change anything there.

**Comment by rochala** [comment]
Right, your error is still suppresed by tryEither...

https://github.com/lampepfl/dotty/blob/91063dd9939ba26f30a538889162b6832cfc2c19/compiler/src/dotty/tools/dotc/ast/Desugar.scala#L1748

`checkWellFormedTupleElems` does not verify match of names.

**Comment by rochala** [comment]
My fix then is not complete, as it will also not report anything from that line, while it should.

**Comment by KacperFKorban** [comment]
We can just treat them as two separate error (which they sort of are).
